src/silx/gui/plot/PlotWidget.py

```diff
@@ -1168,7 +1168,10 @@ def addCurve(self, x, y, legend=None, info=None,
             # Override previous/default values with provided ones
             curve.setInfo(info)
             if color is not None:
-                curve.setColor(color)
+                curve.setColor(
+                    colors.rgba(color, colors=self.getDefaultColors())
+                    if isinstance(color, str) else color
+                )
```

Comment:
Only convert `str` since `color` can also be an array of colors, one for each point..